### ISSUE TITLE
[Tracker] torch.sparse semi-structured 2.3 beta release

### ISSUE THREAD

**Issue by jcaip** [pytorch/pytorch#115662]
This issue tracks the status of semi-structured sparsity to be promoted from prototype status -> beta, targeting the 2.3 release.

* Public APIs / operator support
    * -[x] add in padding support for dense matrices (#110583)
    * -[x] Hook up _FUSE_TRANSPOSE flag to semi-structured sparse subclass
    * -[x] User API for 2:4 sparse training
      * -[x] Merge in subclassing updates from xFormers (#117302)
      * -[x] Upstream custom fast sparsification kernels from xFormers (#122350)
    * -[x] add support for passing in and finding `alg_id` to _cslt_sparse_mm (#115178)
    * -[x] add float32 1:2 sparse support for CUTLASS / cuSPASRELt (#115550)
* Quantization composability
    * -[x] add support for passing in alpha vector of scales into `_cslt_sparse_mm` (#112056)
    * -[x] i8i8 -> fp16 support for `_cslt_sparse_mm` (#109214)
    * -[x] i8i8 -> bf16 support for `_cslt_sparse_mm`
    * -[x] i8i8 -> i32 support for `_cslt_sparse_mm`(#110499)
    * -[x] Add support for CUTLASS fused dequant epllogues (#https://github.com/NVIDIA/cutlass/pull/1189)
    * -[ ] **deprioritized** Add `out_dtype` support for semi-structured sparse subclass (Issue: #115420) 
* PT2 integration
    * -[x] flatten + unflatten functions for `torch.compile` composability (#111049)
    * -[x] meta registrations for `_cslt_sparse_mm` and `_sparse_semi_structured_linear` (#114685)
* Bug Fixes
    * -[x] Issue: #109175  
    * -[x] Issue: #110664 PR: #110420 
    * -[x] Issue: #113776 PR: #114273
    * -[x] Issue: #115154 
* Documentation 
    * -[ ] Update semi-structured sparse docs
    * -[x] Update semi-structured sparse [tutorial](https://pytorch.org/tutorials/advanced/semi_structured_sparse.html)
    * -[ ] 2:4 sparse + quantization tutorial
    * -[x] 2:4 sparse training [docs](https://github.com/pytorch/ao/tree/main/torchao/sparsity/training) and [blog post](https://pytorch.org/blog/accelerating-neural-network-training/)
* Testing / Better Engineering
    * -[x] Split cuSPARSELt and CUTLASS tensor subclasses apart.  
    * -[x] Add cuSPARSELt to conda + pip builds. 
    * -[x] Add cuSPARSELt to CI (#115369)
    * -[ ] Add cuSPASRELt as a backend to PyTorch, easy way to check if cuSPASRELt is installed. (#128534)
    * -[x] Update tests to be parameterized on backend better
   

cc @alexsamardzic @nikitaved @pearu @cpuhrsch @amjames @bhosmer @supriyar @malfet
